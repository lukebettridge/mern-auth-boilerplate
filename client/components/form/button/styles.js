import styled, { css } from "styled-components";

const Button = styled.button`
	background-color: #f8f8f8;
	border: 1px solid #eeeeee;
	cursor: pointer;
	font-size: 13pt;
	font-weight: 600;
	max-width: 350px;
	padding: 8px 20px;
	width: 100%;

	${props =>
		props.as === "a" &&
		css`
			background-color: transparent;
			color: #596f86;
			display: block;
			font-weight: 400;
			max-width: none;
			text-decoration: none;
			&:hover {
				text-decoration: underline;
			}

			${props.secondary &&
				css`
					color: #aaaaaa;
					font-size: 11pt;
					font-weight: 300;
				`}
		`}

	${props =>
		props.mt === "s" &&
		css`
			margin-top: 16px;
		`}

	${props =>
		props.mb === "s" &&
		css`
			margin-bottom: 16px;
		`}
`;

module.exports = { Button };
