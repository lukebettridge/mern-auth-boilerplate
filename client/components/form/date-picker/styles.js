import styled, { css } from "styled-components";

const DatePickerContainer = styled.div`
	${props =>
		props.inError &&
		css`
			& + label {
				color: #db1802;
			}
		`}

	${props =>
		(props.value || props.focused) &&
		css`
			& + label {
				font-size: 95%;
				transform: translate3d(0, -100%, 0);
				opacity: 1;
			}
		`}
`;

const Error = styled.p`
	color: #db1802;
	font-size: 10pt;
	font-weight: 300;
	margin-bottom: 16px;
	margin-top: 0;
	&:first-letter {
		text-transform: capitalize;
	}
`;

const Container = styled.div`
	position: relative;

	${props =>
		props.label &&
		css`
			margin-top: 30px;
		`}

	${props =>
		props.mb &&
		css`
			margin-bottom: 30px;
		`}

	.SingleDatePicker,
	.SingleDatePickerInput,
	.DateInput {
		display: block;
		width: 100%;
	}

	.DateInput_input {
		background-color: transparent;
		border-bottom: 1px solid #e9ecef;
		font-size: 11pt;
		font-weight: 300;
		letter-spacing: 1px;
		margin-bottom: 8px;
		padding: 12px 10px 5px;
		transition: border-color 100ms;
		width: 100%;

		&:hover {
			border-color: #b3b3b3;
		}

		${props =>
			props.inError &&
			css`
				border-color: #db1802;
			`}
	}
`;

const Label = styled.label`
	color: #999999;
	left: 0;
	opacity: 0.5;
	font-weight: 300;
	padding-left: 10px;
	padding-top: 10px;
	position: absolute;
	top: 0;
	transition: all 200ms;
`;

export { Container, DatePickerContainer, Error, Label };
