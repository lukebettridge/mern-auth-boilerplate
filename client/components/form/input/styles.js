import styled, { css } from "styled-components";
import breakpoints from "components/styles/breakpoints";

const Error = styled.p`
	color: #db1802;
	font-size: 10pt;
	font-weight: 300;
	margin-bottom: 16px;
	margin-top: 0;
	&:first-letter {
		text-transform: capitalize;
	}
`;

const Input = styled.input`
	background-color: transparent;
	border: 0;
	border-bottom: 1px solid #e9ecef;
	box-shadow: none;
	font-size: 11pt;
	font-weight: 300;
	margin-bottom: 8px;
	outline: none;
	padding: 12px 10px;
	width: 100%;

	&:focus + label {
		font-size: 75%;
		transform: translate3d(0, -100%, 0);
		opacity: 1;
	}

	${props =>
		props.value &&
		css`
			& + label {
				font-size: 75%;
				transform: translate3d(0, -100%, 0);
				opacity: 1;
			}
		`}

	${props =>
		props.inError &&
		css`
			border-color: #db1802;

			& + label {
				color: #db1802;
			}
		`}
`;

const InputContainer = styled.div`
	position: relative;
	width: 100%;

	@media (min-width: ${breakpoints.m}) {
		min-width: 250px;
	}

	${props =>
		props.label &&
		css`
			margin-top: 20px;
		`}

	${props =>
		props.mb &&
		css`
			margin-bottom: 30px;
		`}
`;

const Label = styled.label`
	left: 0;
	opacity: 0.5;
	font-weight: 300;
	padding-left: 10px;
	padding-top: 10px;
	position: absolute;
	top: 0;
	transition: all 200ms;
`;

export { Error, Input, InputContainer, Label };
