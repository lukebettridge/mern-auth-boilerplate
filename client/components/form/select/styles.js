import styled, { css } from "styled-components";
import breakpoints from "components/styles/breakpoints";
import { default as ReactSelect } from "react-select";

const Error = styled.p`
	color: #db1802;
	font-size: 10pt;
	font-weight: 300;
	margin-bottom: 16px;
	margin-top: 0;
	&:first-letter {
		text-transform: capitalize;
	}
`;

const Select = styled(ReactSelect)`
	&:focus + label {
		font-size: 75%;
		transform: translate3d(0, -100%, 0);
		opacity: 1;
	}

	${props =>
		props.value &&
		props.value.length > 0 &&
		css`
			& + label {
				font-size: 75%;
				transform: translate3d(0, -100%, 0);
				opacity: 1;
			}
		`}

	${props =>
		props.inError &&
		css`
			& + label {
				color: #db1802;
			}
		`}
        
	.react-select__control {
		appearance: none;
		background-color: transparent;
		border: 0;
		border-bottom: 1px solid #e9ecef;
		border-radius: 0;
		box-shadow: none;
		font-size: 12pt;
		font-weight: 300;
		height: 42px;
		margin-bottom: 8px;
		outline: none;
		width: 100%;

		${props =>
			props.inError &&
			css`
				border-color: #db1802;
			`}

		.react-select__input {
			opacity: 0;
		}
	}
`;

const SelectContainer = styled.div`
	position: relative;
	width: 100%;

	@media (min-width: ${breakpoints.m}) {
		min-width: 250px;
	}

	${props =>
		props.label &&
		css`
			margin-top: 30px;
		`}

	${props =>
		props.mb &&
		css`
			margin-bottom: 30px;
		`}
`;

const Label = styled.label`
	left: 0;
	opacity: 0.5;
	font-weight: 300;
	padding-left: 10px;
	padding-top: 10px;
	position: absolute;
	top: 0;
	transition: all 200ms;
`;

export { Error, Select, SelectContainer, Label };
