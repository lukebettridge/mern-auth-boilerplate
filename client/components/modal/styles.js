import styled, { css } from "styled-components";
import breakpoints from "components/styles/breakpoints";

import ReactModal from "react-modal";

const Body = styled.div`
	@media (min-width: ${breakpoints.m}) {
		padding: 10px 20px;
	}

	${props =>
		!props.sideModal &&
		css`
			text-align: center;
		`}
`;

const Close = styled.a`
	height: 32px;
	opacity: 0.3;
	position: relative;
	width: 32px;

	&:hover {
		cursor: pointer;
		opacity: 1;
	}

	&:before,
	&:after {
		background-color: #333;
		content: " ";
		height: 33px;
		left: 15px;
		position: absolute;
		width: 2px;
	}

	&:before {
		transform: rotate(45deg);
	}

	&:after {
		transform: rotate(-45deg);
	}
`;

const Header = styled.div`
	display: flex;
	flex-direction: row-reverse;
	justify-content: space-between;
	margin-bottom: 30px;

	@media (min-width: ${breakpoints.m}) {
		padding-left: 20px;
	}
`;

const Heading = styled.h2`
	font-weight: 700;
	margin: 0;
`;

const Modal = styled(ReactModal)`
	background-color: #ffffff;
	border: 1px solid #e9ecef;
	bottom: auto;
	left: 50%;
	max-width: 650px;
	outline: none;
	padding: 20px 20px 62px;
	position: absolute;
	right: auto;
	top: 43%;
	width: calc(100% - 20px);

	${props =>
		props.sideModal
			? css`
					border: 0;
					border-left: 1px solid #e9ecef;
					bottom: 0;
					left: auto;
					max-width: 720px;
					overflow-y: auto;
					padding: 40px 20px;
					right: -100%;
					top: 0;
					transition: right 0.5s;
					width: 100%;
					z-index: 2;

					&.ReactModal__Content--after-open {
						right: 0;
					}

					@media (min-width: ${breakpoints.m}) {
						padding: 60px;
					}
			  `
			: css`
					border-radius: 2px;
					padding-bottom: 62px;
					transform: translate(-50%, -50%);
			  `}
`;

module.exports = { Body, Close, Header, Heading, Modal };
