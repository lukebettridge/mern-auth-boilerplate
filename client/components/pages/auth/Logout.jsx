import React from "react";
import { Link } from "react-router-dom";
import Cookies from "js-cookie";

const Logout = () => {
	Cookies.remove("authenticated");
	return (
		<div>
			<Link to="/login">Login</Link>
			Successfully logged out.
		</div>
	);
};

export default Logout;
