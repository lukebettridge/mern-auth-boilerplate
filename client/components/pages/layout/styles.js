import styled, { css } from "styled-components";
import breakpoints from "components/styles/breakpoints";

import { Link as RouterLink } from "react-router-dom";

const anchorStyles = `
	align-items: center;
	color: #294661;
	cursor: pointer;
	display: flex;
	font-size: 12pt;
	font-weight: 300;
	justify-content: space-between;
	opacity: 0.6;
	text-decoration: none;
	transition: opacity 0.3s;

	&:hover {
		opacity: 1;
	}

	span svg {
		font-size: 18pt;
		margin-right: 6px;
		padding-bottom: 3px;
		vertical-align: middle;
	}
`;

const subAnchorStyles = `
	${anchorStyles}
	font-size: 11pt;
	padding-left: 15px;
`;

const Anchor = styled.a`
	${anchorStyles}
	${props =>
		props.active &&
		css`
			opacity: 1;
		`}
`;

const Container = styled.div`
	display: flex;
	height: 100%;
`;

const Link = styled(RouterLink)`
	${anchorStyles}
`;

const List = styled.ul`
	margin: 0;
`;

const ListItem = styled.li`
	margin-bottom: 40px;
`;

const Slider = styled.div`
	align-items: center;
	background-color: #fbfbfc;
	border-left: 1px solid #e9ecef;
	border-bottom: 1px solid #e9ecef;
	color: #294661;
	display: flex;
	height: 50px;
	justify-content: center;
	position: absolute;
	right: 0;
	top: 0;
	width: 50px;
	z-index: 1;
	@media (min-width: ${breakpoints.m}) {
		display: none;
	}
`;

const SubAnchor = styled.a`
	${subAnchorStyles}
`;

const SubLink = styled(RouterLink)`
	${subAnchorStyles}
`;

const SubList = styled.ul`
	margin-top: 15px;
	max-height: 1000px;
	overflow: hidden;
	transition: all 0.3s;

	${props =>
		!props.active &&
		css`
			margin: 0;
			max-height: 0;
		`}
`;

const SubListItem = styled.li`
	margin-bottom: 8px;
`;

const Navigation = styled.div`
	background-color: #fbfbfc;
	border-right: 1px solid #e9ecef;
	display: none;
	height: 100%;
	max-width: 200px;
	min-width: 200px;
	padding-left: 15px;
	padding-right: 10px;
	padding-top: 48px;
	width: 200px;
	z-index: 1;

	${props =>
		props.active &&
		css`
			display: table-cell;
			position: absolute;
		`}

	@media (min-width: ${breakpoints.m}) {
		display: table-cell;
		height: unset;
		position: unset;
	}
`;

module.exports = {
	Anchor,
	Container,
	Link,
	List,
	ListItem,
	Navigation,
	Slider,
	SubAnchor,
	SubLink,
	SubList,
	SubListItem
};
