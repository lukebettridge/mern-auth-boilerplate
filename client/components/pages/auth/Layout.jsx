import React from "react";
import PropType from "prop-types";

import { Container, Heading, Paragraph } from "../../styles";

const Layout = props => {
	return (
		<Container small>
			<Heading>Webapp</Heading>
			{props.children}
			<Paragraph center light mt="l">
				Developed by{" "}
				<a
					href="https://paddl.co.uk/"
					rel="noopener noreferrer"
					target="_blank"
				>
					Paddl
				</a>
				<br />. A limited company registered in England and Wales. Registered
				number: 11992253.
			</Paragraph>
		</Container>
	);
};

Layout.propTypes = {
	children: PropType.any
};

export default Layout;
