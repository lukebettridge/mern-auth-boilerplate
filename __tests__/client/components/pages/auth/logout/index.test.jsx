import React from "react";
import { mount } from "enzyme";
import toJSON from "enzyme-to-json";
import cloneDeep from "lodash/cloneDeep";
import { MemoryRouter } from "react-router-dom";
import { MockedProvider } from "@apollo/react-testing";

import { updateWrapper } from "tests/test-utils";

import Layout from "components/pages/auth/Layout";
import { Paragraph } from "components/styles";

import Logout, { ADD_TOKEN_MUTATION } from "components/pages/auth/logout";

const defaultMocks = [
	{
		request: {
			query: ADD_TOKEN_MUTATION
		},
		result: {
			data: { addToken: true }
		}
	}
];
let wrapper;
let subject;

describe("Logout component", () => {
	beforeEach(() => {
		jest.resetAllMocks();
	});

	afterEach(() => {
		if (wrapper) wrapper.unmount();
	});

	it("snapshot renders", async () => {
		mountWrapper();

		await updateSubject(100);

		expect(toJSON(subject)).toMatchSnapshot();
	});

	it.skip("renders component", async () => {
		mountWrapper();

		await updateSubject();

		expect(subject.find(Paragraph).length).toEqual(2);
	});

	it("renders loading", () => {
		mountWrapper();
		expect(subject.find(Layout).length).toEqual(1);
		expect(subject.find(Paragraph).length).toEqual(1);
	});

	it("does not render component on error", async () => {
		const mocks = cloneDeep(defaultMocks);
		Object.assign(mocks[0], { error: new Error() });

		mountWrapper(mocks);

		await updateSubject();

		expect(subject.find(Paragraph).length).toEqual(1);
	});
});

const mountWrapper = (mocks = defaultMocks) => {
	wrapper = mount(
		<MockedProvider addTypename={false} mocks={mocks}>
			<MemoryRouter>
				<Logout />
			</MemoryRouter>
		</MockedProvider>
	);
	subject = wrapper.find(Logout);
};

const updateSubject = async (amount = 0) => {
	await updateWrapper(wrapper, amount);
	subject = wrapper.find(Logout);
};
