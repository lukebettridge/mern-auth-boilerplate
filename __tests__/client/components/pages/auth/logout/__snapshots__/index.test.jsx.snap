// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout component snapshot renders 1`] = `
.c2 {
  background-color: #ffffff;
  border: 1px solid #e9ecef;
  margin-bottom: 20px;
  padding: 40px 30px;
  text-align: center;
}

.c0 {
  margin: 0 auto;
  min-height: 100vh;
  padding: 32px 15px;
  width: 100vw;
  max-width: 540px;
}

.c1 {
  color: #212529;
  font-size: 30pt;
  font-weight: 900;
  -webkit-letter-spacing: -1px;
  -moz-letter-spacing: -1px;
  -ms-letter-spacing: -1px;
  letter-spacing: -1px;
  margin-top: 0;
  margin-bottom: 32px;
}

.c3 {
  color: #666666;
  font-weight: 300;
  line-height: 1.9;
  margin-top: 0;
  margin-bottom: 16px;
  color: #cccccc;
  font-size: 10pt;
  line-height: 1.4;
  text-align: center;
}

.c3 a {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

@media (min-width:768px) {
  .c2 {
    padding: 54px 80px;
    padding-bottom: 50px;
  }
}

@media (min-width:768px) {
  .c0 {
    max-width: calc(100vw - 200px);
    padding: 32px 48px;
  }
}

@media (min-width:768px) {
  .c0 {
    max-width: 700px;
  }
}

<Logout>
  <Layout>
    <styles__Container
      small={true}
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "styles__Container-sc-1upuse9-1",
              "isStatic": false,
              "lastClassName": "c0",
              "rules": Array [
                "margin:0 auto;min-height:100vh;padding:32px 15px;width:100vw;@media (min-width:",
                "768px",
                "){max-width:calc(100vw - 200px);padding:32px 48px;}",
                [Function],
                " ",
                [Function],
                " ",
                [Function],
              ],
            },
            "displayName": "styles__Container",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "styles__Container-sc-1upuse9-1",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        small={true}
      >
        <div
          className="c0"
        >
          <styles__Heading>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "styles__Heading-sc-1upuse9-6",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "color:#212529;font-size:30pt;font-weight:900;letter-spacing:-1px;margin-top:0;margin-bottom:32px;",
                    ],
                  },
                  "displayName": "styles__Heading",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "styles__Heading-sc-1upuse9-6",
                  "target": "h1",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <h1
                className="c1"
              >
                Webapp
              </h1>
            </StyledComponent>
          </styles__Heading>
          <styles__Box>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "styles__Box-sc-1upuse9-0",
                    "isStatic": false,
                    "lastClassName": "c2",
                    "rules": Array [
                      "background-color:#ffffff;border:1px solid #e9ecef;margin-bottom:20px;padding:40px 30px;text-align:center;@media (min-width:",
                      "768px",
                      "){padding:54px 80px;padding-bottom:50px;}",
                      [Function],
                    ],
                  },
                  "displayName": "styles__Box",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "styles__Box-sc-1upuse9-0",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="c2"
              >
                <Mutation
                  mutation={
                    Object {
                      "definitions": Array [
                        Object {
                          "directives": Array [],
                          "kind": "OperationDefinition",
                          "name": Object {
                            "kind": "Name",
                            "value": "addToken",
                          },
                          "operation": "mutation",
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "addToken",
                                },
                                "selectionSet": undefined,
                              },
                            ],
                          },
                          "variableDefinitions": Array [],
                        },
                      ],
                      "kind": "Document",
                      "loc": Object {
                        "end": 36,
                        "start": 0,
                      },
                    }
                  }
                  onError={[Function]}
                />
              </div>
            </StyledComponent>
          </styles__Box>
          <styles__Paragraph
            center={true}
            light={true}
          >
            <StyledComponent
              center={true}
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "styles__Paragraph-sc-1upuse9-8",
                    "isStatic": false,
                    "lastClassName": "bQhQRy",
                    "rules": Array [
                      "color:#666666;font-weight:300;line-height:1.9;margin-top:0;margin-bottom:16px;",
                      [Function],
                      " ",
                      [Function],
                      " a{color:inherit;text-decoration:none;&:hover{text-decoration underline;}}",
                    ],
                  },
                  "displayName": "styles__Paragraph",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "styles__Paragraph-sc-1upuse9-8",
                  "target": "p",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              light={true}
            >
              <p
                className="c3"
              >
                Developed by
                 
                <a
                  href="https://paddl.co.uk/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Paddl
                </a>
                <br />
                . A limited company registered in England and Wales. Registered number: 11992253.
              </p>
            </StyledComponent>
          </styles__Paragraph>
        </div>
      </StyledComponent>
    </styles__Container>
  </Layout>
</Logout>
`;
